packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -350,94 +353,101 @@ const _BaseButton: React.ForwardRefRenderFunction<BladeElementRef, BaseButtonPro
   });
 
   return (
-    <StyledBaseButton
-      ref={buttonRef}
-      accessibilityProps={{ ...makeAccessible({ role: 'button', label: accessibilityLabel }) }}
-      isLoading={isLoading}
-      disabled={disabled}
-      activeBorderColor={activeBorderColor}
-      activeBackgroundColor={activeBackgroundColor}
-      defaultBorderColor={defaultBorderColor}
-      minHeight={minHeight}
-      buttonPaddingTop={buttonPaddingTop}
-      buttonPaddingBottom={buttonPaddingBottom}
-      buttonPaddingLeft={buttonPaddingLeft}
-      buttonPaddingRight={buttonPaddingRight}
-      defaultBackgroundColor={defaultBackgroundColor}
-      focusBorderColor={focusBorderColor}
-      focusBackgroundColor={focusBackgroundColor}
-      focusRingColor={focusRingColor}
-      hoverBorderColor={hoverBorderColor}
-      hoverBackgroundColor={hoverBackgroundColor}
-      isFullWidth={isFullWidth}
-      onClick={onClick}
-      type={type}
-      borderWidth={borderWidth}
-      borderRadius={borderRadius}
-      motionDuration={motionDuration}
-      motionEasing={motionEasing}
-      {...metaAttribute(MetaConstants.Component, MetaConstants.Button)}
+    // Marking this inline-block because button is an inline-block element and we don't want to give it styles of Box
+    <BaseBox
+      display={isReactNative() ? undefined : 'inline-block'}
+      width={isFullWidth ? '100%' : 'auto'}
+      {...getStyledProps(styledProps)}
```

Comment:
Do we need a wrapper? Can't we add styled props directly to `StyledBaseButton`? The idea to introduce styled props was to avoid wrapper hell but if we're also adding wrappers then won't the final outcome be the same?

Reply:
https://github.com/razorpay/blade/pull/1016#issuecomment-1446015208

Reply:
I think we should still avoid adding extra wrapper for any components. This would hamper our approach to keep the HTML as clean as possible. If we notice some issues like the `order` prop which you mentioned on svg, we can always move the wrapper approach but by default I think we should avoid adding it

Reply:
Have removed wrappers from most elements now